README.md

```diff
@@ -226,6 +229,70 @@ The only required key/value in the signing_info dictionary is 'identity'.
 See the **SIGNED PACKAGES** section of the man page for `pkgbuild` or the **SIGNED PRODUCT ARCHIVES** section of the man page for `productbuild` for more information on the signing options.
 
 
+### Package notarization
+
+**Important notes**:
+
+- Please read the [Customizing the Notarization Workflow](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution/customizing_the_notarization_workflow) web page before you start notarizing your packages.
+- Xcode 10 (or newer) is **required**.  If you have more than one version of Xcode installed on your Mac, be sure to use the xcode-select utility to choose the appropriate version: `sudo xcode-select -s /path/to/Xcode10.app`.
+- Unproxied network access to the Apple infrastruture (Usually `17.0.0.0/8` network) is required.
+- Notarization tool tries to notarize not only the package but also the package payload. All code in the payload (including but not limited to app bundles, frameworks, kernel extensions) needs to be properly signed with the hardened runtime restrictions in order to be notarized. Please read Apple Developer documentation for more information.
+
+You may notarize **SIGNED PACKAGES** as part of the build process by adding a `notarization_info` dictionary to the build\_info.plist:
+
+```plist
+    <key>notarization_info</key>
+    <dict>
+        <key>username</key>
+        <string>john.appleseed@apple.com</string>
+        <key>password</key>
+        <string>@keychain:AC_PASSWORD</string>
+        <key>asc_provider</key>
+        <string>JohnAppleseed1XXXXXX8</string>
+        <key>staple_timeout</key>
+        <integer>600</integer>
+    </dict>
+```
+
+or, in JSON format in a build-info.json file:
+
+```json
+    "notarization_info": {
+        "username": "john.appleseed@apple.com",
+        "password": "@keychain:AC_PASSWORD",
+        "asc_provider": "JohnAppleseed1XXXXXX8",
+        "stapler_timeout": 600
+    }
+```
+
+Keys/values of the `notarization_info` dictionary:
+
+| Key               | Type    | Required | Description |
+| ----------------- | ------- | -------- | ----------- |
+| username          | String  | Yes      | Login email adress of your developer Apple ID |
+| password          | String  | Yes      | 2FA app specific password. For information about the password and saving it to the login keychain see the web page [Customizing the Notarization Workflow](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution/customizing_the_notarization_workflow) |
+| asc_provider      | String  | No       | Only needed when a user account is associated with multiple providers |
+| primary_bundle_id | String  | No       | Defaults to `identifier`. `primary_bundle_id` is usefull when `identifier` contains characters such as '_' Apple notary service does not like |
```

Comment:
I think providing this option is fine, but since we know that Apple's notary service does not like underscores in the primary bundle id, if this option is not provided and you/we "auto-generate" the primary bundle id from the pkg id, why don't we automatically substitute hyphens for underscores?

Reply:
Is basic `string.replace()` like this 736741c ok?

Reply:
See my comment on that commit.